cartographer/io/internal/mapping_state_serialization.cc

```diff
@@ -60,9 +60,86 @@ mapping::proto::SerializationHeader CreateHeader() {
   return header;
 }
 
-SerializedData SerializePoseGraph(const mapping::PoseGraph& pose_graph) {
+SerializedData SerializePoseGraph(const mapping::PoseGraph& pose_graph,
+                                  bool include_unfinished_submaps) {
   SerializedData proto;
   *proto.mutable_pose_graph() = pose_graph.ToProto();
+
+  if (include_unfinished_submaps) {
+    return proto;
+  }
+
+  std::set<mapping::SubmapId> unfinished_submaps;
+  std::set<mapping::NodeId> orphaned_nodes;
+
+  // Determine set of unfinished submaps.
```

Comment:
I am undecided whether this logic should be located in `PoseGraph` rather or here. If located in `PoseGraph`, couldn't you easily work on non-proto types and then serialize?